server/src/routes/api/v1/ingredientsRouter.js

```diff
@@ -0,0 +1,83 @@
+import cleanUserInput from "../../../services/cleanUserInput.js"
+const { ValidationError, NotNullViolationError } = objection
+
+const ingredientsRouter = new express.Router()
+
+ingredientsRouter.get("/list/:userId", async (req, res) => {
+  const { userId } = req.params
+  try {
+    const user = await User.query().findById(userId)
+    const ingredients = await user.$relatedQuery("ingredients")
+    const serializedIngredients = ingredients.map(ingredient => {
+      return IngredientSerializer.getIngredientInfo(ingredient)
+    })
+
+    return res.status(200).json({ ingredients: serializedIngredients })
+  } catch(err) {
+    return res.status(500).json({ err })
+  }
+})
+
+ingredientsRouter.get("/:id", async (req, res) => {
+  const { id } = req.params
+  try {
+    const ingredient = await Ingredient.query().findById(id)
+    const serializedIngredient = await IngredientSerializer.getIngredientWithPantryMeasurements(ingredient)
+    return res.status(200).json({ ingredient: serializedIngredient})
+  } catch(err) {
+    return res.status(500).json({ err })
+  }
+})
+
+ingredientsRouter.post("/", async (req, res) => {
+  const { newIngredient, userId } = req.body
+  const cleanedIngredient = cleanUserInput(newIngredient)
+
+  try {
+    const existingIngredient = await Ingredient.query().findOne({ name: cleanedIngredient.name })
+    if (existingIngredient) {
+
+      const ingredientInPantry = await User.relatedQuery("ingredients").for(userId).where({ name: cleanedIngredient.name })
+      if (ingredientInPantry.length > 0) {
+        return res.status(422).json({ 
+          errors: {
+            "Looks like": [{
+              message: "you already own this ingredient!  Please edit that ingredient instead of creating a new one." 
+            }]
+          }
+        })
+      }
+
+      const relatedIngredient = await User.transaction(async (trx) => {
+        const user = await User.query(trx).findById(userId)
+        const relatedIngredient = await user.$relatedQuery("ingredients", trx).relate({
+          id: existingIngredient.id,
+          amount: cleanedIngredient.amount,
+          unit: cleanedIngredient.unit,
+          description: cleanedIngredient.description
+        })
+        return relatedIngredient
+      })
+      console.log(relatedIngredient)
+      return res.status(201).json("Successful input")
+    }
+
+    const fetchedIngredient = await User.transaction(async (trx) => {
+      const user = await User.query(trx).findById(userId)
+      const fetchedIngredient = await user.$relatedQuery("ingredients", trx).insertAndFetch(cleanedIngredient)
+      return fetchedIngredient
+    })
+    return res.status(201).json("successful input")
+  } catch(err) {
+    if (err instanceof ValidationError || err instanceof NotNullViolationError) {
+      return res.status(422).json({ errors: err.data})
+    }
+    return res.status(500).json({ err })
+  }
+})
+
+export default ingredientsRouter
```

Comment:
its worth mentioning that even though all of your ingredients are now in one table (ownedIngredients + recipeIngredients) you actually could still have a separate `userIngredients` api router for when you are making requests for ingredients for a specific user. I know this would require more refactoring, but would probably be better practice and keep you organized based on the CRUD you planning on doing in the future